docs/semgrep-secrets/validators.md

```diff
@@ -0,0 +1,327 @@
+---
+slug: validators
+title: Validators
+hide_title: true
+description: Learn about validators used in Semgrep Secrets rules.
+tags:
+  - Semgrep Secrets
+---
+
+# Validators
+
+[Semgrep Secrets](/semgrep-secrets/conceptual-overview) uses proprietary **validators** to determine if a secret is
+actively being used. Validators are included in the
+[rules](/semgrep-secrets/rules) that Semgrep Secrets uses.
+
+This article walks you through the syntax required to write your own custom
+validators.
+
+:::note Validation
+Semgrep currently supports validation using HTTP and HTTPS.
+:::
+
+## Sample validator
+
+```yaml
+validators:
+- http:
+    request:
+      headers:
+        Authorization: Bearer $REGEX
+        Host: api.exampleCo.com
+        User-Agent: Semgrep
+      method: GET
+      url: https://api.exampleCo.com/user
+    response:
+    - match:
+      - status-code: 200
+      result:
+        validity: valid
+    - match:
+      - status-code: 401
+      result:
+        validity: invalid 
+```
+
+<details>
+<summary>See a validator in the context of a full rule.</summary>
+
+```yaml
+rules:
+- id: exampleCo_example
+  message: >-
+    This is an example rule that performs validation against exampleCo.com
+  severity: WARNING 
+  metadata:
+    product: secrets
+    secret_type: exampleCo
+  languages:
+  - regex
+  validators:
+  - http:
+      request:
+        headers:
+          Authorization: Bearer $REGEX
+          Host: api.exampleCo.com
+          User-Agent: Semgrep
+        method: GET
+        url: https://api.exampleCo.com/user
+      response:
+      - match:
+        - status-code: 200
+        result:
+          validity: valid
+      - match:
+        - status-code: 401
+        result:
+          validity: invalid 
+  patterns:
+  - patterns:
+    - pattern-regex: (?<REGEX>\b(someprefix_someRegex[0-9A-Z]{32})\b)
+    - focus-metavariable: $REGEX
+    - metavariable-analysis:
+        analyzer: entropy
+        metavariable: $REGEX
+```
+
+</details>
+
+## Syntax
+
+### validator
+
+| Key | Required | Description |
+| - | - | - |
+| validator | Yes | Used to define a validator within a Semgrep rule. |
+
+### http
+
+| Key | Required | Description |
+| - | - | - |
+| http | Yes | Indicates that the request type is `http`. |
```

Comment:
Not a huge fan of the wording of this. It implies to me that we may add additional keys, row-wise (i.e., like additional structure/product fields), but it would be other mutually exclusive keys, like ftp or ssh, etc...